File: cpp/p45.cpp
#include "euler.h"

using namespace std;

int main() {
  int n = 286;
	
  while( !is_s_gonal( get_s_gonal( n, 3 ), 5 ) || 
	 !is_s_gonal( get_s_gonal( n, 3 ), 6 ) )
    n++;

  cout << get_s_gonal( n, 3 ) << endl;
}

File: clojure/p45.clj
(ns p45
  (:use [euler :only (s-gonal? triangles timed-test)]))

(defn penta-hexa-gonal? [n]
  (and (s-gonal? 5 n) (s-gonal? 6 n)))

(timed-test
 1533776805
 (let [tri-penta-hexa-gonals (filter penta-hexa-gonal? (triangles))]
   (nth tri-penta-hexa-gonals 2)))
